translator/translate/otel/receiver/prometheus/translator.go

```diff
@@ -0,0 +1,100 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package prometheus
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/receiver"
+	"gopkg.in/yaml.v3"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
+)
+
+const (
+	otelConfigParsingError = "has invalid keys: global"
+	defaultTlsCaPath       = "/etc/amazon-cloudwatch-observability-agent-cert/tls-ca.crt"
+)
+
+var (
+	configPathKey = common.ConfigKey(common.PrometheusConfigKeys[component.DataTypeMetrics], common.PrometheusConfigPathKey)
+)
+
+type translator struct {
+	name    string
+	factory receiver.Factory
+}
+
+type Option func(any)
+
+var _ common.Translator[component.Config] = (*translator)(nil)
+
+func NewTranslator(opts ...Option) common.Translator[component.Config] {
+	t := &translator{factory: prometheusreceiver.NewFactory()}
+	for _, opt := range opts {
+		opt(t)
+	}
+	return t
+}
+
+func (t *translator) ID() component.ID {
+	return component.NewIDWithName(t.factory.Type(), t.name)
+}
+
+func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
+	cfg := t.factory.CreateDefaultConfig().(*prometheusreceiver.Config)
+
+	if conf.IsSet(configPathKey) {
+		configPath, _ := common.GetString(conf, configPathKey)
+		content, err := os.ReadFile(configPath)
+		if err != nil {
+			return nil, fmt.Errorf("unable to read prometheus config from path: %w", err)
+		}
+
+		var stringMap map[string]interface{}
+		err = yaml.Unmarshal(content, &stringMap)
+		if err != nil {
+			return nil, err
+		}
+		componentParser := confmap.NewFromStringMap(stringMap)
+		if componentParser == nil {
+			return nil, fmt.Errorf("unable to parse config from filename %s", configPath)
+		}
+		var isPrometheusConfig bool
+		err = componentParser.Unmarshal(&cfg)
+		if err != nil {
+			// passed in prometheus config is in plain prometheus format and not otel wrapper
+			if !strings.Contains(err.Error(), otelConfigParsingError) {
+				return nil, fmt.Errorf("unable to unmarshall config to otel prometheus config from filename %s", configPath)
+			}
+			isPrometheusConfig = true
+		} else {
+			// given prometheus config is in otel format so check if target allocator is being used
+			// then add the default cert for TargetAllocator
+			if cfg.TargetAllocator != nil && len(cfg.TargetAllocator.CollectorID) > 0 {
+				cfg.TargetAllocator.TLSSetting.Config.CAFile = defaultTlsCaPath
+				cfg.TargetAllocator.TLSSetting.ReloadInterval = 10 * time.Second
```

Comment:
Even though we set it here I don't see it in generated yaml: 
```
receivers:
    prometheus:
        config:
            global:
                evaluation_interval: 1m
                scrape_interval: 1m
                scrape_protocols:
                    - OpenMetricsText1.0.0
                    - OpenMetricsText0.0.1
                    - PrometheusText0.0.4
                scrape_timeout: 10s
        report_extra_scrape_metrics: false
        start_time_metric_regex: ""
        target_allocator:
            collector_id: cloudwatch-agent-w-amp-1
            compression: ""
            disable_keep_alives: false
            endpoint: https://target-allocator-service:80
            http2_ping_timeout: 0s
            http2_read_idle_timeout: 0s
            interval: 30s
            proxy_url: ""
            read_buffer_size: 0
            timeout: 0s
            tls:
                ca_file: /etc/amazon-cloudwatch-observability-agent-cert/tls-ca.crt
                cert_file: ""
                include_system_ca_certs_pool: false
                insecure: false
                insecure_skip_verify: false
                key_file: ""
                max_version: ""
                min_version: ""
                reload_interval: 0s
                server_name_override: ""
            write_buffer_size: 0
        trim_metric_suffixes: false
        use_start_time_metric: false
``` 
Could we add unit-tests for this to make sure

Reply:
There is unit test for this already https://github.com/aws/amazon-cloudwatch-agent/pull/1399/files#diff-3834305eb1507b10bb2f8d589bff9f3ce7824a1b1b0a646e573297965e3aa61cR84